app/models/concerns/user/omniauth.rb

```diff
@@ -9,16 +9,23 @@ def add_data_from_facebook(auth)
 
   class_methods do
     def from_omniauth(auth)
-      user = where(provider: auth.provider, uid: auth.uid).first_or_initialize do |user|
+      find_by(provider: auth.provider, uid: auth.uid) ||
+        find_by(email: auth.info.email) ||
```

Comment:
isn't it some security issue?

Reply:
Do you mean that it will be possible to hijack an account if an attacker knows Facebook App ID?

Reply:
@dmitryzuev Searched for this on the stackoverflow and looks like it common approach to the problem. I realized that I miss one thing here - if a user found by email it can be a good idea to fill `provider` and `uid` for him.

Reply:
filling `provider` and `uid` will do, i think